classes/blocks/class-spreadsheet.php

```diff
@@ -42,49 +41,12 @@ public function __construct() {
 	}
 
 	/**
-	 * Get all the data that will be needed to render the block correctly.
+	 * Required by the `Base_Block` class.
 	 *
-	 * @param array $fields This is the array of fields of this block.
-	 *
-	 * @return array The data to be passed in the View.
+	 * @param array $fields Unused, required by the abstract function.
```

Comment:
I'm usually not a fan of long inheritance chains, but maybe it's more appropriate here if we create sub-types for each kind of block? So that would make
```
|Base_Block
|\
||Static_Block
|\
||Dynamic|Block
```

Reply:
Not proposing to do this in this PR btw :D probably best handled separately.

Reply:
Then we could just move the `prepare_data` function to `Dynamic_Block`.

Reply:
Indeed, this is calling for a few refactors.